### ISSUE TITLE
ESLint looks for .eslintrc.* files even when `-c` flag is used

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#9535]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 8.8.0
* **npm Version:** 5.4.2

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
# foo.yml

rules:
  semi: 2
```

```yaml
# .eslintrc.yml

rules:
  no-console: 2
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```
console.log('foo');
```

```bash
eslint -c foo.yml foo.js
```

**What did you expect to happen?**

I expected no errors to be reported, because I used the `-c` flag to specify a config, so I did not expect ESLint to also look for other config files in the filesystem.

**What actually happened? Please include the actual, raw output from ESLint.**

```
1:1   error  Unexpected console statement         no-console
```

In addition to the config file specified with `-c`, ESLint also looked for `.eslintrc` files in the current directory, so the `no-console` rule was enabled.

**Comment by alberto** [comment]
Does it also happen with `root: true`?

**Comment by gyandeeps** [comment]
I dont think its a bug, it was designed this way. You have to use `--no-eslintrc` option to disable that feature. So `-c` and `--no-eslintrc` are individual options.

`root: true` is used inside `.eslintrc` files to stop the traversal lookup of `.eslintrc` files up in the hierarchy chain.

**Comment by platinumazure** [comment]
I agree with @gyandeeps, this is as designed. We could update the docs to clarify that it might make sense to also use `--no-eslintrc` if no other config files should be used.

I do use this feature in combination with ESLint config files, so please let's not change this if possible...

**Comment by platinumazure** [comment]
I'd be okay with changing this in a major release if it's possible to combine with a `--eslintrc` or `--use-eslintrc` to explicitly opt back into normal config resolution. But I want to continue supporting the use case (eslintrc resolution plus overrides in `-c`).

**Comment by not-an-aardvark** [comment]
Ok, it seems like this is working as designed, so I'll update it to be a documentation issue. I think we should document more clearly that `-c` still causes ESLint to look for other config files.

**Comment by WalterBoles** [comment]
Ok are they my pull request

**Comment by WalterBoles** [comment]
Is it fixed and can I get some held don't want there to be anyway to get into my account except biometric finger prints and facial recognition and eye printing if there is a seventeen digital number ss# then birthdate

**Comment by platinumazure** [comment]
@WalterBoles I've created #9847 to address this, which should be merged later this week and will be updated on eslint.org when we cut a new minor release on Friday or Saturday.
